packages/react/src/Testimonial/Testimonial.test.tsx

```diff
@@ -0,0 +1,516 @@
+        <Testimonial.Name>Name</Testimonial.Name>
+        <Testimonial.Avatar src={mockAvatarSrc} alt={mockAvatarAlt} size={64} />
+      </Testimonial>,
+    )
+
+    const avatarContainer = getByAltText(mockAvatarAlt)
+    expect(avatarContainer).toBeInTheDocument()
+  })
+
+  it('renders logo with img element', () => {
+    const logoSrc = '/logo.png'
+    const logoAlt = 'Company logo'
+
+    const {getByRole} = render(
+      <Testimonial>
+        <Testimonial.Quote>Quote text</Testimonial.Quote>
+        <Testimonial.Name>Name</Testimonial.Name>
+        <Testimonial.Logo>
+          <img src={logoSrc} alt={logoAlt} />
+        </Testimonial.Logo>
+      </Testimonial>,
+    )
+
+    const logo = getByRole('img', {name: logoAlt})
+    expect(logo).toBeInTheDocument()
+    expect(logo).toHaveAttribute('src', logoSrc)
+    expect(logo).toBeInTheDocument()
+  })
+
+  it('renders logo container', () => {
+    const {getByRole} = render(
+      <Testimonial>
+        <Testimonial.Quote>Quote text</Testimonial.Quote>
+        <Testimonial.Name>Name</Testimonial.Name>
+        <Testimonial.Logo>
+          <img src="/logo.png" alt="Logo" />
+        </Testimonial.Logo>
+      </Testimonial>,
+    )
+
+    const logoContainer = getByRole('img', {name: 'Logo'}).closest('div')
+    expect(logoContainer).toBeInTheDocument()
+  })
+
+  it('supports RefObject for Testimonial.Logo component', () => {
+    const refObject = React.createRef<HTMLImageElement>()
+    const {getByRole} = render(
+      <Testimonial>
+        <Testimonial.Quote>Quote text</Testimonial.Quote>
+        <Testimonial.Name>Name</Testimonial.Name>
+        <Testimonial.Logo ref={refObject}>
+          <img src="/logo.png" alt="Logo" />
+        </Testimonial.Logo>
+      </Testimonial>,
+    )
+
+    const img = getByRole('img', {name: 'Logo'})
+    expect(refObject.current).toBe(img)
+  })
+
+  it('supports functional ref for Testimonial.Logo component', () => {
+    const mockRef = jest.fn()
+    const {getByRole} = render(
+      <Testimonial>
+        <Testimonial.Quote>Quote text</Testimonial.Quote>
+        <Testimonial.Name>Name</Testimonial.Name>
+        <Testimonial.Logo ref={mockRef}>
+          <img src="/logo.png" alt="Logo" />
+        </Testimonial.Logo>
+      </Testimonial>,
+    )
+
+    const img = getByRole('img', {name: 'Logo'})
+    expect(mockRef).toHaveBeenCalledWith(img)
+  })
+
+  it('renders quote before media section', () => {
+    const {getByRole, getByText} = render(<DefaultTestimonial />)
+
+    const figure = getByRole('figure')
+    const quote = getByText(mockQuote)
+    const name = getByText(mockName)
+
+    const figureChildren = Array.from(figure.children)
+    const quoteIndex = figureChildren.findIndex(child => child.contains(quote))
+    const nameIndex = figureChildren.findIndex(child => child.contains(name))
+
+    expect(quoteIndex).toBeLessThan(nameIndex)
+  })
+
+  it('renders avatar, logo, and name in media section', () => {
```

Comment:
This one feels a little like implementation detail. It seems like we're expecting elements to render in a specific, generic div but not sure if that's necessary. wdyt?

Reply:
Yep happy to remove it 👍